python-lib/nlp_utils.py

```diff
@@ -3,16 +3,45 @@
 import unicodedata
 
 
-def get_keyword(text: AnyStr, normalize_case: bool) -> AnyStr:
-    """Return text in its wanted-case form"""
-    return text.lower() if normalize_case else text
+def normalize_case_text(text: AnyStr, lowercase: bool) -> AnyStr:
+    """Return text in its wanted case form"""
+    return text.lower() if lowercase else text
 
 
-def get_sentence(span: Span, normalize_case: bool) -> Union[Span, Doc]:
-    """Return Span object as a Doc if case_insensitive is set to True"""
-    return span if normalize_case else span.as_doc()
+def get_token_attribute(lemmatize: bool) -> AnyStr:
+    """Return the attribute to pass to spaCy Matcher"""
+    return "LEMMA" if lemmatize else "TEXT"
 
 
-def normalize_text(texts: List[AnyStr]) -> List[AnyStr]:
+def lemmatize_doc(doc: Doc) -> AnyStr:
+    """Lemmatize SpaCy.tokens.Doc object.
+
+    Args:
+        doc (SpaCy.tokens.Doc): Text to lemmatize
+
+    Returns:
+        str: Text in its lemmatized form
+
+    """
+    return " ".join([span.lemma_ for span in doc])
+
+
+def lemmatize_span(span: Span, lemmatize: bool) -> AnyStr:
+    """Lemmatize SpaCy.tokens.Span object if 'lemmatize' is True.
+
+    Args:
+        span (spacy.tokens.Span): Text to process
+        lemmatize (bool): if True, return the text lemmatized
+
+    Returns:
+        str: Text or lemma associated to the span
+
+    """
+    if lemmatize:
+        return span.lemma_
+    return span.text
+
+
+def unicode_normalize_text(texts: List[AnyStr]) -> List[AnyStr]:
```

Comment:
Depending on what flexibility we want to have later on (using other normalizations), I would either:
- Rename the function to `unicode_normalize_NFD`
- Put a `norm` argument which defaults to `NFD`

If we are sure we'll stick with NFD, ignore this comment

Reply:
Good point ! I have already change this method in PR #19,  so I'd rather let this concern outside of this PR. Here it uses unicode normalization only for the output columns naming. Is it fine if I don't change it in this PR ? :)

Reply:
Sure